diff --git a/src/SDL_shadercross.c b/src/SDL_shadercross.c
index 99ea040..e8c3f7d 100644
--- a/src/SDL_shadercross.c
+++ b/src/SDL_shadercross.c
@@ -397,7 +397,7 @@ static void *SDL_ShaderCross_INTERNAL_CompileUsingDXC(
         defineStringsUtf16[i] = (wchar_t *)SDL_iconv_string("WCHAR_T", "UTF-8", defineString, MAX_DEFINE_STRING_LENGTH);
     }
 
-    LPCWSTR *args = SDL_malloc(sizeof(LPCWSTR) * (numDefineStrings + 13));
+    LPCWSTR *args = SDL_malloc(sizeof(LPCWSTR) * (numDefineStrings + 14));
     Uint32 argCount = 0;
 
     for (Uint32 i = 0; i < numDefineStrings; i += 1) {
@@ -427,15 +427,17 @@ static void *SDL_ShaderCross_INTERNAL_CompileUsingDXC(
 
     if (info->shader_stage == SDL_SHADERCROSS_SHADERSTAGE_VERTEX) {
         args[argCount++] = (LPCWSTR)L"-T";
-        args[argCount++] = (LPCWSTR)L"vs_6_0";
+        args[argCount++] = (LPCWSTR)L"vs_6_2";
     } else if (info->shader_stage == SDL_SHADERCROSS_SHADERSTAGE_FRAGMENT) {
         args[argCount++] = (LPCWSTR)L"-T";
-        args[argCount++] = (LPCWSTR)L"ps_6_0";
+        args[argCount++] = (LPCWSTR)L"ps_6_2";
     } else { // compute
         args[argCount++] = (LPCWSTR)L"-T";
-        args[argCount++] = (LPCWSTR)L"cs_6_0";
+        args[argCount++] = (LPCWSTR)L"cs_6_2";
     }
 
+    args[argCount++] = (LPCWSTR)L"-enable-16bit-types";
+
     if (spirv) {
         args[argCount++] = (LPCWSTR)L"-spirv";
         args[argCount++] = (LPCWSTR)L"-fspv-flatten-resource-arrays";
